clean.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+usage()
+{
+    echo "Usage: clean [options]"
+    echo "Cleans the local dev environment."
+    echo
+    echo "  -b     Delete bin directory"
+    echo "  -p     Delete packages directory and NuGet package caches"
+    echo "  -a     Remove all untracked files in the working tree and delete NuGet package caches"
+    exit 1
+}
+
+# Parse arguments
+
+if [ $# == 0 ]
+then
+    echo "Please specify at least one option."
+    echo "Use 'clean -h' for help."
+    exit 1
+fi
+
+while [[ $# > 0 ]]
+do
+    opt="$1"
+    case $opt in
+        -h|--help)
+        usage
+        ;;
+        -b)
+        CleanBinDir=true
+        ;;
+        -p)
+        CleanPackagesDir=true
+        CleanPackagesCache=true
+        ;;
+        -a)
+        CleanWorkingTree=true
+        CleanPackagesCache=true
+        ;;
+        *)
+        echo "Unrecognized argument '$opt'"
+        echo "Use 'clean -h' for help."
+        exit 1
+        ;;
+    esac
+    shift
+done
+
+WorkingTreeRoot="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+if [ "$CleanBinDir" == true ] && [ "$CleanWorkingTree" != true ]
+then
+    echo "Removing bin directory"
+    rm -rf $WorkingTreeRoot/bin
+fi
+
+if [ "$CleanPackagesDir" == true ] && [ "$CleanWorkingTree" != true ]
+then
+    echo "Removing packages directory"
+    rm -rf $WorkingTreeRoot/packages
```

Comment:
> the Clean MSBuild target is that it has a single set behavior

Well, we could give it more behavior if we'd like to, or add different names, etc.

> it's possible that at some point we'll modify it to also remove packages (for example), in which case we'd also need to update clean.sh

I guess this is accurate, but I think it would really just amount to mapping a couple of arguments, i.e. `clean.sh -p` -> `msbuild /t:clean;cleanpackages`, or whatever we'd call it. You mentioned that we will have a `clean.cmd` as well, so wouldn't that involve duplicating all of this logic there, again? That seems like more work to me than just having both `clean.sh` and `clean.cmd` delegate to msbuild, since we would only have to implement the actual cleaning logic once in msbuild.

I can see how doing things that way might be too much overhead, though, since the clean scripts are fairly simple and small to begin with, so the duplication isn't that bad even in the worst case. I'm just trying to see if we can't try to minimize that duplication without too much trouble.